### PR TITLE
fix(core): skip cache cleanup when enableGlobalCache is true

### PR DESCRIPTION
**Pull request by stevenxu-db**
**What's the problem this PR addresses?**
Skip cache cleanup when `enableGlobalCache` is true and global cache is used.

Discussion: https://discord.com/channels/226791405589233664/226793713722982400/964222577721610280

**How did you fix it?**
Read the config value for current project and short-circuit the cleanup task if set.

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

**Comment by stevenxu-db** [comment]
Please double check my version bumps. First time.

File: packages/yarnpkg-core/sources/Project.ts [hunk 1]
```diff
@@ -1601,7 +1601,7 @@ export class Project {
     await opts.report.startTimerPromise(`Fetch step`, async () => {
       await this.fetchEverything(opts);
 
-      if ((typeof opts.persistProject === `undefined` || opts.persistProject) && opts.mode !== InstallMode.UpdateLockfile) {
+      if ((typeof opts.persistProject === `undefined` || opts.persistProject) && opts.mode !== InstallMode.UpdateLockfile && !this.configuration.get(`enableGlobalCache`)) {
```
[the remark this answers is not in the document]
Comment (stevenxu-db) [reply packages/yarnpkg-core/sources/Project.ts]: Done. I did not find integration test coverage between `yarn test:unit packages/yarnpkg-core/tests/Project.test.ts` and `yarn build:cli && yarn test:integration` for the `persistProject` guard. Let me know if you want me to try to backfill coverage for this with the move, though I'm not intending the change the logic... just refactor it. 

The last touch to this code path was https://github.com/yarnpkg/berry/commit/d13213280191f064785cf0bc19c92c76c60581bd which appears to have some test coverage through the broader code path in `packages/acceptance-tests/pkg-tests-specs/sources/commands/__snapshots__/install.test.js.snap` but the integration test case doesn't include cache cleanup.

File: packages/yarnpkg-core/sources/Project.ts [hunk 2]
```diff
@@ -1838,7 +1835,12 @@ export class Project {
     }
   }
 
-  async cacheCleanup({cache, report}: InstallOptions)  {
+  async cacheCleanup({cache, mode, persistProject, report}: InstallOptions)  {
+    if (mode === InstallMode.UpdateLockfile)
```
Comment (stevenxu-db) [inline packages/yarnpkg-core/sources/Project.ts:1839]: tbh I stared at this for a while trying to invert the guard clause. I think I got it right and am expressing the semantics clearly here, but please double check my work.

File: packages/yarnpkg-core/sources/Project.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (merceyz) [reply packages/yarnpkg-core/sources/Project.ts]: I should have been clearer but I only meant the `this.configuration.get(`enableGlobalCache`)` part of the check

[the remark this answers is not in the document]
Comment (stevenxu-db) [reply packages/yarnpkg-core/sources/Project.ts]: Sounds good. Done.

**Review by merceyz — DISMISSED** [review]
LGTM, thanks!

**Pull request merged**